src/parser/parser_parse.c

```diff
@@ -6,24 +6,33 @@
 /*   By: aruzafa- <aruzafa-@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2023/09/30 17:14:54 by aruzafa-          #+#    #+#             */
-/*   Updated: 2023/12/04 16:07:02 by aruzafa-         ###   ########.fr       */
+/*   Updated: 2023/12/04 20:30:56 by aruzafa-         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "cub3d.h"
 
+static int cub3d_filled(t_cub3d *cub3d)
+{
+	return (cub3d->ceil_color != 0 && cub3d->floor_color
+		&& cub3d->east_texture && cub3d->north_texture
+		&& cub3d->south_texture && cub3d->west_texture
+	);
+}
+
 static t_cub3d	*parse_fd(int fd, t_error *error)
 {
 	t_cub3d	*cub3d;
 	char	*line;
 
 	cub3d = (t_cub3d *) ft_calloc(1, sizeof(t_cub3d));
 	*error = NO_ERROR;
-	line = spaces_trim(ft_get_next_line(fd));
-	while (line && *error == NO_ERROR && !validate_line_is_map(line))
+	while (*error == NO_ERROR && !cub3d_filled(cub3d))
```

Comment:
He cambiado la estrategia para decidir cuando empiezo a validar el mapa.

Antes leía la línea y si es mapa, salía del WHILE para empezar a validar el mapa, pero al haber leído la primera línea, ya la descartaba y la validación del mapa empezaba en la segunda.

En su lugar he decidido comprobar si ya está rellena la estructura del cub3d. En caso de que esté empezamos la validación del mapa.

Esto provoca:
 - Si repito líneas de NO, SU, WE, EA, F, C... Me lo entiende como caracteres inválidos del mapa

Reply:
Arreglado